src/modules/getBullsAndCows.js

```diff
@@ -13,7 +13,35 @@
  * Example: { bulls: 1, cows: 2 }
  */
 function getBullsAndCows(userInput, numberToGuess) {
-  /* Write your code here */
+  const userInputStr = userInput.toString();
+  const numberToGuessStr = numberToGuess.toString();
+
+  const result = {
+    bulls: 0,
+    cows: 0,
+  };
+
+  const checkedPositions = new Array(4).fill(false);
+
+  for (let i = 0; i < 4; i++) {
+    if (userInputStr[i] === numberToGuessStr[i]) {
+      result.bulls++;
+      checkedPositions[i] = true;
+    }
+  }
+
+  for (let i = 0; i < 4; i++) {
+    if (userInputStr[i] !== numberToGuessStr[i]) {
+      const cowIndex = numberToGuessStr.indexOf(userInputStr[i]);
+
+      if (cowIndex !== -1 && !checkedPositions[cowIndex]) {
+        result.cows++;
+        checkedPositions[cowIndex] = true;
```

Comment:
The logic for calculating cows is incorrect. The current implementation checks for the presence of a digit in the `numberToGuessStr` but does not ensure that the cowIndex is not already a bull. This can lead to incorrect cow counts. Consider checking if `cowIndex` is not equal to `i` and that `checkedPositions[cowIndex]` is false before incrementing the cow count.